src/controllers/notesController.js

```diff
@@ -0,0 +1,103 @@
+import mongooseModel from "../db/schemas/notesSchema.js";
+
+const notesController = {
+  get_all: async (req, res) => {
+    await mongooseModel
+      .find({})
+      .then((result) => {
+        res.json(result);
+      })
+      .catch((err) => {
+        if (err instanceof SyntaxError) {
+          console.error("Error:", err.message);
+          res.send("Error: Wrong type of data, cannot get.");
+        } else {
+          console.error("Error updating note:", err.message);
+          res.send("Error: Something went wrong with getting all the data.");
+        }
+      });
+  },
+
+  post: async (req, res) => {
+    const createTitle = req.body.title;
+    const createContent = req.body.content;
+    const createDate = Date.now();
+
+    await mongooseModel
+      .create({
+        title: createTitle,
+        content: createContent,
+        createAt: createDate,
+        updateAt: "",
+      })
+      .then((createdData) => {
+        res.send(createdData);
+      })
+      .catch((err) => {
+        if (err instanceof SyntaxError) {
+          console.error("Error:", err.message);
+          res.send("Error: Wrong type of data, cannot post.");
+        } else {
+          console.error("Error:", err.message);
+          res.send("Error: Something went wrong, couldn't post the data.");
+        }
+      });
+  },
+
+  put: async (req, res) => {
+    const updateId = req.params.id;
+    const updateNote = {
+      title: req.body.title,
+      content: req.body.content,
+      date: Date.now(),
+    };
+
+    await mongooseModel
+      .findOneAndUpdate({ _id: updateId }, { $set: updateNote }, { new: true })
+      .exec()
+      .then((data) => {
+        if (data === null) {
+          res.send("nothing found");
+        } else {
+          res.send(data);
+        }
+      })
+      .catch((err) => {
+        if (err instanceof SyntaxError) {
+          console.error("Error:", err.message);
+          res.send("Error: Wrong type of data, cannot put.");
+        } else {
+          console.error("Error updating note:", err.message);
+          res.send(
+            "Error: Something went wrong, couldn't putting up a new data."
+          );
+        }
+      });
+  },
+
+  delete: async (req, res) => {
+    const deleteId = req.params.id;
+
+    await mongooseModel
+      .findOneAndDelete({ _id: deleteId })
```

Comment:
In production, it's a good practice to use _soft delete_ algorithm: not to remove object at all, but mark it as deleted. To do it, you will need to create an additional field: `isDeleted` with default value: false. 
After that update get request with `deleted: false`, and update deleteNote request with marking note as deleted (isDeleted: true).

Why so? It's important to get all the documents and organize clear structure of the client's flow in case of any errors, in case of getting statistics or issues inside the application/system.